unity-renderer/Assets/DCLPlugins/DebugPlugins/ShapesCollidersDisplayer/DebugShapesCollidersDisplayer.cs

```diff
@@ -0,0 +1,108 @@
+using System.Collections.Generic;
+using System.Linq;
+using DCL;
+using DCL.Controllers;
+using DCLPlugins.DebugPlugins.Commons;
+
+public class DebugShapesCollidersDisplayer : IPlugin
+{
+    private readonly IBaseDictionary<string, bool> isCollidersDisplayEnabledForScene;
+    private readonly WorldRuntimeContext worldRuntime;
+    internal readonly Dictionary<string, WatchSceneHandler> scenesWatcher = new Dictionary<string, WatchSceneHandler>();
+    internal readonly List<string> pendingScenesId = new List<string>();
+
+    public DebugShapesCollidersDisplayer() : this(
+        DataStore.i.debugConfig.showSceneColliders,
+        Environment.i.world) { }
+
+    internal DebugShapesCollidersDisplayer(
+        IBaseDictionary<string, bool> isCollidersDisplayEnabledVariable,
+        WorldRuntimeContext worldRuntime)
+    {
+        this.isCollidersDisplayEnabledForScene = isCollidersDisplayEnabledVariable;
+        this.worldRuntime = worldRuntime;
+
+        // NOTE: we search for scenes that might be added to the variable previous to this class instantiation
+        using (var iterator = isCollidersDisplayEnabledVariable.Get().GetEnumerator())
+        {
+            while (iterator.MoveNext() && iterator.Current.Value)
+            {
+                IsCollidersDisplayEnabledVariableOnOnAdded(iterator.Current.Key, iterator.Current.Value);
+            }
+        }
+
+        isCollidersDisplayEnabledVariable.OnAdded += IsCollidersDisplayEnabledVariableOnOnAdded;
+        isCollidersDisplayEnabledVariable.OnRemoved += IsCollidersDisplayEnabledVariableOnOnRemoved;
+        worldRuntime.sceneController.OnNewSceneAdded += SceneControllerOnOnNewSceneAdded;
+    }
+    public void Dispose()
+    {
+        isCollidersDisplayEnabledForScene.OnAdded -= IsCollidersDisplayEnabledVariableOnOnAdded;
+        isCollidersDisplayEnabledForScene.OnRemoved -= IsCollidersDisplayEnabledVariableOnOnRemoved;
+        worldRuntime.sceneController.OnNewSceneAdded -= SceneControllerOnOnNewSceneAdded;
+
+        var scenesId = scenesWatcher.Keys.ToArray();
+        for (int i = 0; i < scenesId.Length; i++)
+        {
+            KillWatchScene(scenesId[i]);
+        }
+    }
+
+    private void KillWatchScene(string sceneId)
```

Comment:
Kill sounds a bit misleading since you are not killing the scene to watch, maybe RemoveWatchScene or StopWatchScene??